Make Dynamo inline through torch._library.custom_ops.autograd

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135066

Fixes https://github.com/pytorch/pytorch/issues/135057

The bug was: in the situation that Dynamo graph breaks in the forward
and Compiled Autograd uses Dynamo to introspect the backward, we end up
running into a "Unsupported: inlining through SKIPFILES" error. The
solution is to mark the entirety of this module as inlineable.

Test Plan:
- new test

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @rec

zou3519 committed:
Make Dynamo inline through torch._library.custom_ops.autograd

Fixes https://github.com/pytorch/pytorch/issues/135057

Test Plan:
- new test

[ghstack-poisoned]

yf225 commented on test/inductor/test_compiled_autograd.py:
```diff
@@ -172,6 +172,26 @@ def fn():
 
         self.check_output_and_recompiles(fn)
 
+    def test_graph_break_custom_op(self):
```
curious: does it mean we need to graph-break when custom op is used in compiled autograd graph?

(Traceable FSDP2 requires that compiled autograd backward graph has no graph break, so hoping graph-break here is not a requirement haha)

zou3519 replied:
> curious: does it mean we need to graph-break when custom op is used in compiled autograd graph?

No. We are testing what happens if (1) a user has a custom op that graph breaks in the forward and (2) we use compiled autograd on this

zou3519 committed:
Update on "Make Dynamo inline through torch._library.custom_ops.autograd"


Fixes https://github.com/pytorch/pytorch/issues/135057

The bug was: in the situation that Dynamo graph breaks in the forward
and Compiled Autograd uses Dynamo to introspect the backward, we end up
running into a "Unsupported: inlining through SKIPFILES" error. The
solution is to mark the entirety of this module as inlineable.

Test Plan:
- new test

[ghstack-poisoned]

zou3519 committed:
Update on "Make Dynamo inline through torch._library.custom_ops.autograd"


Fixes https://github.com/pytorch/pytorch/issues/135057

The bug was: in the situation that Dynamo graph breaks in the forward
and Compiled Autograd uses Dynamo to introspect the backward, we end up
running into a "Unsupported: inlining through SKIPFILES" error. The
solution is to mark the entirety of this module as inlineable.

Test Plan:
- new test

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge -f "unrelated failures"
